account_move_journal_template/models/account_move.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2022 Therp BV <https://therp.nl>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from odoo import _, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    journal_entry_ids = fields.One2many(
+        comodel_name="account.move", inverse_name="journal_entry_id",
+    )
+    journal_entry_id = fields.Many2one(comodel_name="account.move")
+
+    def button_draft(self):
+        res = super().button_draft()
+        if self.journal_entry_ids:
+            self.journal_entry_ids.button_draft()
+        return res
+
+    def button_cancel(self):
+        res = super().button_cancel()
+        if self.journal_entry_ids:
+            self.journal_entry_ids.button_cancel()
+        return res
+
+    def action_view_journal_entries(self):
+        self.ensure_one()
+        return {
+            "name": _("Journal Entries"),
+            "type": "ir.actions.act_window",
+            "res_model": "account.move",
+            "view_mode": "tree,form",
+            "target": "current",
+            "domain": [("id", "in", self.journal_entry_ids.ids)],
+        }
+
+    def post(self):
+        res = super().post()
+        all_journals = self.env["account.move"]
+        for move in self:
+            if move.type not in ["out_invoice", "out_refund"]:
+                continue
+            # Remove previous lines
```

Comment:
Why only the lines? And maybe, if each post action results in a full regeneration, should lines (and moves) not be deleted / ulinked when setting an invoice to draft?

Reply:
Better keep the moves here, and add/remove lines between state transitions. if an extra move is cancelled or in draft, it won't partake into anything really. User can delete them if they feel like it.